OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -934,7 +934,144 @@ internal static void UpgradeActors(int engineVersion, ref List<MiniYamlNode> nod
 		{
 			foreach (var node in nodes)
 			{
-				// Add rules here
+				// Fix RA building footprints to not use _ when it's not necessary
+				if (engineVersion < 20160530 && Game.ModData.Manifest.Mod.Id == "ra")
+				{
+					// Tesla Coil
+					if (node.Value.Value == "tsla")
```

Comment:
This is indeed for the resource center and other third party maps.  It sucks to have these checks hardcoded in common code, but we can't fix that until we redo the upgrade rules plumbing.